### PR TITLE
ember-select-light integration

### PR DESCRIPTION
**Pull request by joe-wroten-q2**
@abdulmhamid @pixelhandler replacing native select with `ember-select-light` for proper dog fooding. Open to feedback.

### CODE REVIEW

File: tests/integration/components/select-box-test.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe('Integration | Component | select box', function () {
         this.set('valueKey', 'value');
         this.set('options', [{value:'1', display: 'first'}, {value:'2', display: 'second'}, {value:'3', display: 'third'}, {value:'4', display: 'fourth'}]);
         this.render(hbs `{{select-box options=options displayKey=displayKey valueKey=valueKey nativeSelect=true}}`);
-        expect(findAll('[data-test-id^="sbNativeDeepOption"]')).to.have.lengthOf(4);
```
Comment (pixelhandler) [inline tests/integration/components/select-box-test.js]: I'm curious if there is still a way we can use test ids. Using the elements to select in the test seems fine though.

**Review by pixelhandler — APPROVED** [review]

File: addon/templates/components/select-box.hbs [hunk 2]
```diff
@@ -5,18 +5,17 @@
...
-					<option value={{get option valueKey}} selected={{eq selected (get option valueKey)}} data-test-id={{concat "sbNativeDeepOption" index}}>{{get option displayKey}}</option>
-				{{else}}
-					<option value={{option}} selected={{eq selected option}} data-test-id={{concat "sbNativeFlatOption" index}}>{{option}}</option>
-				{{/if}}
-			{{/each}}
-		</select>
+		{{select-light
+			name=selectId
+			id=selectId
+			class="mdl-textfield__input"
```
Comment (abdulmhamid) [inline addon/templates/components/select-box.hbs:11]: Note: we are getting rid of this class in favor of something more dynamic.

File: tests/integration/components/select-box-test.js (hunk 1 above)
Comment (abdulmhamid) [reply tests/integration/components/select-box-test.js]: Does `select-light` have a className assignment? If so, we can use that if keeping the test id is not an option.  I'd like for test selectors to be more specific than element names.

**Comment by joe-wroten-q2** [comment]
@abdulmhamid `selectId` used for sturdier test element targeting. Will leave it up to you to merge and bump should you approve.

**Comment by abdulmhamid** [comment]
@joe-wroten-q2 Do you intend for `selectId` to be string value passed in explicitly or a dynamic value using the guid (e.g. `unified-select-2324`)?

**Comment by joe-wroten-q2** [comment]
`selectId` was an available property that already existed and it paired up nicely with `ember-select-light`'s name and id support. So it's just meant to be a static id override for testing purposes, no need for it to be more complicated in this case.

**Pull request merged**